crates/runc/src/lib.rs

```diff
@@ -340,11 +339,16 @@ impl Runc {
     }
 
     /// Update a container with the provided resource spec
-    pub fn update(&self, id: &str, resources: &Linux) -> Result<()> {
-        let filename = utils::temp_filename_in_runtime_dir()?;
-        let spec_json =
-            serde_json::to_string(resources).map_err(Error::JsonDeserializationFailed)?;
-        std::fs::write(&filename, spec_json).map_err(Error::SpecFileCreationFailed)?;
+    pub fn update(&self, id: &str, resources: &LinuxResources) -> Result<()> {
+        let (mut temp_file, filename) = utils::make_temp_file_in_runtime_dir()?;
+        {
+            let f = temp_file.as_file_mut();
+            let spec_json =
+                serde_json::to_string(resources).map_err(Error::JsonDeserializationFailed)?;
+            f.write(spec_json.as_bytes())
+                .map_err(Error::SpecFileCreationFailed)?;
+            f.flush().map_err(Error::SpecFileCreationFailed)?;
```

Comment:
Should we extract to helper func for writing to a temp file?

Reply:
Solved